assets/src/hooks/useNotifications.ts

```diff
@@ -1,12 +1,21 @@
 import { Socket } from "phoenix"
+import { array, Infer, object, union } from "superstruct"
 import {
   NotificationData,
   notificationFromData,
 } from "../models/notificationData"
 import { Notification } from "../realtime.d"
-import { useChannel } from "./useChannel"
+import { useCheckedChannel } from "./useChannel"
 
-type InitialNotificationData = { initial_notifications: NotificationData[] }
+const InitialNotificationData = object({
+  initial_notifications: array(NotificationData),
+})
+type InitialNotificationData = Infer<typeof InitialNotificationData>
+
+const ReceivedNotificationsData = union([
+  NotificationData,
+  InitialNotificationData,
+])
```

Comment:
That's a good point. We haven't always been very good about the organization of this stuff (whether or not a given interface gets defined in `modules/`, whether the `Foo` and `FooData` interfaces are defined in the same module or separate modules...). This case feels a little weird and it's sort of a consequence of the fact that the channel receives two different kinds of data, one on join and one on push (in contrast to all of our other channels). I'll give some thought to whether or not there's a better place to put this particular code.

Reply:
So I'm actually thinking that in this particular case it's best to leave it in `useNotifications.ts` since it's so tightly coupled to how the channel is used and they don't even need to be exported for other modules. Does that make sense? I agree that in general we could be more disciplined about putting things in `models/` if they need to be shared across different parts of the code.

Reply:
That seems reasonable to me!